tubular/salesforce_api.py

```
@@ -0,0 +1,91 @@
+            raise Exception("Could not find Salesforce user with email %s" % assignee_email)
+
+    def _get_id_by_email(self, object_type, email):
+        """
+        Given an id, query for an <object_type> with that email
+        Returns the id of that object or None if not found
+        Raises an exception if multiple are found
+        """
+        id_query = self._sf.query("SELECT Id FROM %s WHERE Email='%s'" % (object_type, email))
+        total_size = int(id_query['totalSize'])
+        if total_size == 0:
+            return None
+        elif total_size == 1:
+            return id_query['records'][0]['Id']
+        else:
+            raise Exception("Multiple results for email %s" % email)
+
+    def get_lead_id_by_email(self, email):
+        """
+        Given an id, query for an <object_type> with that email
+        Returns the id of that object or None if not found
+        Raises an exception if multiple are found
+        """
+        return self._get_id_by_email("Lead", email)
+
+    def get_user_id_by_email(self, email):
+        """
+        Given an email, returns the user id for the User with that email
+        or None if no user is found
+        """
+        return self._get_id_by_email("User", email)
+
+    def _create_retirement_task(self, email, lead_id):
+        """
+        Creates a Salesforce Task instructing a user to manually retire the
+        given lead
+        """
+        task_params = {
+            'Description': RETIREMENT_TASK_DESCRIPTION % email,
+            'Subject': "GDPR Request: " + email,
+            'WhoId': lead_id,
+            'OwnerId': self.assignee_id
+        }
+        created_task = self._sf.Task.create(task_params)
+        if created_task['success']:
+            log.info("Successfully salesforce task created task " + created_task['id'])
+        else:
+            log.error("Errors while creating task:")
+            for error in created_task['errors']:
+                log.error(error)
```

We should raise an error here as we will need to manually reset and re-run the user if this doesn't succeed.